src/calculateAverage.js

```diff
@@ -37,7 +37,40 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === undefined) {
+      return this.reduce((a, b) => a + b) / this.length;
+    } else if (options === null
+      || typeof (options) === 'string'
+      || typeof (options.accumulator) === 'string'
+      || options.length > 1) {
+      return undefined;
+    }
+
+    let sum = 0;
+    let counter = 0;
+
+    for (let i = 0; i < this.length; i++) {
+      if (options.accumulator !== undefined
+        && typeof (options.accumulator) === 'function') {
+        sum += options.accumulator(this[i], i, this);
+        counter++;
+        continue;
+      }
+
+      if (options.propertyName === undefined
+        && this[i][options.propertyName] === undefined) {
+        continue;
+      }
+
+      if (typeof (this[i][options.propertyName]) !== 'number') {
+        continue;
+      } else {
+        sum += this[i][options.propertyName];
+        counter++;
+      }
+    }
```

Comment:
если в этом случаи тебе нужен один из четырех вариантов, то можно объединить в if else
```suggestion
      if (options.accumulator !== undefined
        && typeof (options.accumulator) === 'function') {
        sum += options.accumulator(this[i], i, this);
        counter++;
      } else if (options.propertyName === undefined
        && this[i][options.propertyName] === undefined) {
        continue;
      } else if (typeof (this[i][options.propertyName]) !== 'number') {
        continue;
      } else {
        sum += this[i][options.propertyName];
        counter++;
      }
    }
```